src/macros.rs

```diff
@@ -72,6 +72,62 @@ macro_rules! decl_derive {
     };
 }
 
+/// The `decl_attribute!` macro declares a custom attribute wrapper. It will
+/// parse the incoming `TokenStream` into a `synstructure::Structure` object,
+/// and pass it into the inner function.
+///
+/// Your inner function should have the following type:
+///
+/// ```
+/// # extern crate quote;
+/// # extern crate proc_macro2;
+/// # extern crate synstructure;
+/// fn attribute(
+///     attr: proc_macro2::TokenStream,
+///     structure: synstructure::Structure,
+/// ) -> proc_macro2::TokenStream {
+///     unimplemented!()
+/// }
+/// ```
+///
+/// # Usage
+///
+/// ```
+/// # #[macro_use] extern crate quote;
+/// # extern crate proc_macro2;
+/// # extern crate synstructure;
+/// # fn main() {}
+/// fn attribute_interesting(
+///     _attr: proc_macro2::TokenStream,
+///     _structure: synstructure::Structure,
+/// ) -> proc_macro2::TokenStream {
+///     quote! { ... }
+/// }
+///
+/// # const _IGNORE: &'static str = stringify! {
+/// decl_attribute!([interesting] => attribute_interesting);
+/// # };
+/// ```
+/// ```
+#[macro_export]
+macro_rules! decl_attribute {
+    // XXX: Switch to using this variant everywhere?
```

Comment:
Im not sure what you're referring to with this comment?

Reply:
Leftover from copy-pasting `decl_derive`.